tests/unit/workflows/test_workflows_tasks_submission.py

```diff
@@ -747,3 +747,253 @@ def test_prepare_keywords_does_nothing_if_no_keywords_were_predicted():
 
     assert validate(result['keywords'], subschema) is None
     assert expected == result['keywords']
+
+
+def test_send_robotupload_doesnt_stop_authors_if_update_to_legacy_is_disabled():
+    with requests_mock.Mocker() as requests_mocker:
+        requests_mocker.register_uri(
+            'POST', 'http://inspirehep.net/batchuploader/robotupload/insert',
+            text='[INFO] foo bar baz'
+        )
+
+        config = {
+            'LEGACY_ROBOTUPLOAD_URL': 'http://inspirehep.net',
+            'PRODUCTION_MODE': True,
+            'FEATURE_FLAG_ENABLE_UPDATE_TO_LEGACY': False
+        }
+
+        with patch.dict(current_app.config, config), \
+                patch('inspirehep.modules.workflows.tasks.submission.record2marcxml') as mock_record2marcxml:
+            data = {
+                '$schema': 'http://localhost:5000/schemas/records/authors.json',
+                'name': {
+                    'preferred_name': 'Jessica Jones',
+                    'value': 'Jones, Jessica'
+                }
+            }
+            extra_data = {}
+
+            obj = MockObj(data, extra_data)
+            obj.workflow.name == 'author'
+            eng = MockEng()
+
+            _send_robotupload = send_robotupload(
+                mode='insert',
+            )
+
+            assert _send_robotupload(obj, eng) is None
+            assert mock_record2marcxml.called_with(data)
```

Comment:
don't you rather want to assert that the request to legacy has been done instead of mocking `record2marcxml`? same in all tests.